Modules/Module_Processing/Module_ASL/xASL_wrp_RealignASL.m

```diff
@@ -506,30 +506,22 @@ function xASL_wrp_RealignASL(x, bASL)
         TempIm = xASL_io_Nifti2Im(InputPath);
         
         % Remove spikes
-        NewIm = TempIm(:,:,:, ~exclusion);
-        
         % skip this for fMRI, which is more complicated
         % due to tissue T1 effects (incomplete saturation, so temporal relation between volumes)
-        % Save in single precision, since conversion back to INT16 would
-        % otherwise loose precious precision
-        % PM: can this be simplified to same format as original??
-        xASL_io_SaveNifti(x.P.Path_ASL4D, x.P.Path_despiked_ASL4D, NewIm, 32, 0);
+        NewIm = TempIm(:,:,:, ~exclusion);
         
         % Do same for *.mat motion sidecars
         LoadParms = load(x.P.Path_ASL4D_mat, '-mat');
         mat = LoadParms.mat;
-        
         mat = mat(:,:, ~exclusion);
-
         save(x.P.Path_despiked_ASL4D_mat, 'mat');
 
         % Also change parameter vectors
-        x.Q.EchoTime = x.Q.EchoTime(~exclusion);
-        x.Q.Initial_PLD = x.Q.Initial_PLD(~exclusion);
-        x.Q.LabelingDuration = x.Q.LabelingDuration(~exclusion);
-
-        % #997 Need to save this in json sidecar for further usage in the pipeline
+        jsonFields.EchoTime = x.Q.EchoTime(~exclusion);

```

Comment:
The function parms2BIDS will be able to handle that. But still, it's better to use the correct format and write
jsonFields.Q.EchoTime instead of jsonFields.EchoTime - and the same for the other two fields.

Reply:
Done in 6608345d8747fad9f022083eed435441de6c2bb5